src/Program.cs

```diff
@@ -32,7 +32,7 @@ public static void Main(string[] args)
                 var xmlKey = File.ReadAllText("PublicKey.xml");
                 var rsa = new RSACryptoServiceProvider();
                 rsa.FromXmlString(xmlKey);
-                return rsa.VerifyData(computedSignature, archive.Signature, HashAlgorithmName.SHA1, RSASignaturePadding.Pkcs1);
+                return rsa.VerifyData(computedSignature, archive.Signature, archive.Metadata.HashAlgorithmName, RSASignaturePadding.Pkcs1);
```

Comment:
Now it will start failing on old archives, because their metadata doesn't contain `HashAlgorithmName` (is it serializable, by the way?).

Also, we should probably do it based on version? We could say that version 4.0 uses SHA-256 and 1.x uses SHA-1?